architecture-decisions/pages/multiple-brokers.md

```diff
@@ -0,0 +1,88 @@
+ADR: Multiple Brokers
+=============
+
+Features & Approach
+-------------
+
+| Feature | Redis | RabbitMQ | Amazon SQS |
+| --- | --- | --- | --- |
+| **Enqueue** | `LPUSH` | `JOB-QUEUE(durable: true)`<br>`PUBLISH(persistent: true)` | `QUEUE(type: FIFO)`<br>`SEND` |
```

Comment:
Are we sure that we want this to be FIFO only (in SQS case)? FIFO queues (at least in SQS) also provide exactly-once guarantees, hence they are more expensive to run and are also significantly capped on quotas (80% less in-flight capacity).

I think the decision of FIFO or not should be left up to the users to configure.

Sidekiq cannot also guarantee this, since multiple jobs are processed in parallel (same with goose, today). [Ref](https://github.com/mperham/sidekiq/wiki/FAQ#how-can-i-process-a-certain-queue-in-serial).

Reply:
It's worth documenting, as an overall design, under what circumstances ordering is guaranteed, and when not.

Reply:
Along with, it would be good to document semantics around:

* Delivery
* Reliability
* Retryability
* Burying / revival

For each broker.